src/config.rs

```diff
@@ -204,6 +210,32 @@ impl Config {
         self.search = Some(search);
         self.last_search = LastSearch::Search;
     }
+
+    /// Normalize config.
+    pub fn glibc_normalize(&mut self) {
+        self.nameservers.truncate(NAMESERVER_LIMIT);
+        self.search = self.search.take().map(|mut s| {
+            s.truncate(SEARCH_LIMIT); 
+            s
+        });
+    }
+
+    /// Get nameserver or on the local machine
+    pub fn get_nameserver_or_local(&self) -> Vec<ScopedIp> {
+        if self.nameservers.is_empty() {
+            vec![
+                ScopedIp::from(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1))), 
+                ScopedIp::from(IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1))),
+            ]
+        } else {
+            self.nameservers.to_vec()
+        }
+    }
+
+    /// Get domain or hostname
+    pub fn get_domain_or_hostname(&self) -> Option<String> {
```

Comment:
I'm not sure but I think I would do `get_domain_from_env`, and that function would look env variable first and than these two things. The idea is that `env` used here is a broad sense. Although may be there is a better name?

Or do you think there is a use case for getting domain without looking at env variables?

Reply:
you mean `LOCALDOMAIN`? 

this env varibale is for `search` keyword.
> The search keyword of a system's resolv.conf file can be overridden
       on a per-process basis by setting the environment variable
       LOCALDOMAIN to a space-separated list of search domains.

Reply:
Okay, I was confused by env name :)

Okay, still not sure about function name. Something like `compute_domain` maybe.

Reply:
Or `get_system_domain` ? (especially if we name feature gate `system`)